code/lib/preview-web/src/View.ts

```diff
@@ -0,0 +1,24 @@
+import type { Store_Story } from '@storybook/types';
+
+export abstract class View<TStorybookRoot> {
+  // Get ready to render a story, returning the element to render to
+  abstract prepareForStory(story: Store_Story<any>): TStorybookRoot;
+
+  abstract prepareForDocs(): TStorybookRoot;
+
+  abstract showErrorDisplay(err: { message?: string; stack?: string }): void;
+
+  abstract showNoPreview(): void;
+
+  abstract showPreparingStory(options: { immediate: boolean }): void;
+
+  abstract showPreparingDocs(): void;
+
+  abstract showMain(): void;
+
+  abstract showDocs(): void;
+
+  abstract showStory(): void;
+
+  abstract showStoryDuringRender(): void;
+}
```

Comment:
I recon this is to accommodate more use-cases? react-native?
Why have this class structure where 1 class extends the other?

I assume considerable thought has gone into it, but I have a bit of an aversion to classes and class inheritance.

Reply:
It could be an interface if you prefer that @ndelangen?

Reply:
Yeah, I think I'd prefer that, but this is not a blocker IMHO